components/AddANote.vue

```diff
@@ -1,55 +1,46 @@
 <script setup>
-import { computed, ref } from "vue";
-import { usePackageState } from "../composables";
+import { createToast } from "mosha-vue-toastify";
+import { computed, ref, watchEffect } from "vue";
+import { useSingleReport } from "../composables";
 
 const props = defineProps({
-  defaultNote: {
-    type: String,
-    required: false,
-    default: "",
-  },
   name: {
     type: String,
     required: true,
   },
 });
 
 const tempFormValue = ref("");
-const { addNote } = usePackageState();
+const { addNote, report } = useSingleReport();
 
-const noteInnerText = computed(() => {
-  return tempFormValue.value || props.defaultNote;
+watchEffect(() => {
+  tempFormValue.value = report.value?.state?.[props.name]?.note;
 });
 
 const noteText = computed(() =>
-  props.defaultNote
+  tempFormValue.value
     ? "update note for this library"
     : "add a note for this library"
 );
 
-const updateFormText = (evt) => {
-  tempFormValue.value = evt.target.value;
-};
-
 const notePlaceholder = "add note";
 const buttonText = ref("+");
 
-/*
-  Right now this just updates state —
-  we should make it actually save and give feedback
-*/
-const saveNote = () => {
+const saveNote = async () => {
   buttonText.value = "...";
-  addNote({ pkg: props.name, note: tempFormValue.value });
+  await addNote({ pkg: props.name, note: tempFormValue.value });
```

Comment:
**suggestion:** If we keep the two values as noted above, we should be able to do the update here with the return now that you (**praise**) updated things so that we don't need to pass down the prop.

Reply:
Replied in nabove message